scripts/ci-e2e.sh

```diff
@@ -3,21 +3,36 @@
 set -euxo pipefail
 
 REPO_ROOT=$(realpath "$(dirname "$(realpath "${BASH_SOURCE[0]}")")"/..)
-cd "${REPO_ROOT}"
+cd "${REPO_ROOT}" || exit 1
 export CAPM3PATH="${REPO_ROOT}"
-export WORKING_DIR=/opt/metal3-dev-env
+export WORKING_DIR=/tmp/metal3-dev-env
 FORCE_REPO_UPDATE="${FORCE_REPO_UPDATE:-false}"
 
 export CAPM3RELEASEBRANCH="${CAPM3RELEASEBRANCH:-main}"
 
+# Verify they are available and have correct versions.
+PATH=$PATH:/usr/local/go/bin
+PATH=$PATH:$(go env GOPATH)/bin
+
+"${REPO_ROOT}/hack/ensure-go.sh"
+# shellcheck source=./hack/ensure-kind.sh
+source "${REPO_ROOT}/hack/ensure-kind.sh"
```

Comment:
```suggestion
. "${REPO_ROOT}/hack/ensure-kind.sh"
```
Same for below lines as well.